### DISCUSSION TITLE
Ensure Valid File Suffix for `CommunityToolkit.Maui` and `CommunityToolkit.Maui.Core`

### DISCUSSION THREAD

**Discussion by TheCodeTraveler** [CommunityToolkit/Maui#316] [General]
We are looking for a solution to ensure that invalid files suffixes for files in `CommunityToolkit.Maui` and `CommunityToolkit.Maui.Core` generate a compiler error .

As noted in https://github.com/CommunityToolkit/Maui/pull/315, a couple files with invalid suffixes snuck into the `main` branch.

The following are valid Suffixes for `CommunityToolkit.Maui` and `CommunityToolkit.Maui.Core`; all other suffixes are invalid:
- `*.shared.cs`
  - This file is compiled for each target framework, similar to a [Shared Project](https://docs.microsoft.com/xamarin/cross-platform/app-fundamentals/shared-projects?WT.mc_id=mobile-0000-bramin)
- `*.ios.cs
  - This file is compiled for `net6.0-ios`
- `*. macos.cs
  - This files is compiled for `net6.0-maccatalyst`
- `*.macios.cs`
  - This file is compiled for `net6.0-ios` and `net6.0-maccatalyst`
- `*.android.cs`
  - This file is compiled for `net6.0-android`
- `*.windows.cs`
  - This file is compiled for `net6.0-windows`

**Comment by VladislavAntonyuk** [comment]
The first thing that comes to my mind is using the pre-commit hook with a bash script.
I don't know if Roslyn analyzer can do it

**Comment by CliffAgius** [comment]
Closed as merged. #315
